cfme/infrastructure/pxe.py

```diff
@@ -156,15 +254,32 @@ def create(self, cancel=False, refresh=True, refresh_timeout=120):
             refresh (boolean): Whether to run the refresh operation on the PXE server after
                 the add has been completed.
         """
-        navigate_to(self, 'Add')
-        fill(pxe_properties_form, self._form_mapping(True, **self.__dict__))
-        self._submit(cancel, form_buttons.add)
-        if not cancel:
-            flash.assert_message_match('PXE Server "{}" was added'.format(self.name))
+        view = navigate_to(self, 'Add')
+        view.fill({'name': self.name,
```

Comment:
I was thinking of this and came to conclusion that it was bad idea to click cancel when there were not changes. If user wants to create or update object and that object doesn't have actual changes,  test should fail exactly in the place of error. However, if method does cancel, the test will go further.

btw, thank you reminder, I'll remove changed in the rest of places